Voycar.Api.Web.Tests.Integration/Users/Post_Register/Tests.cs

```diff
@@ -0,0 +1,327 @@
+            FirstName = "null",
+            LastName = "null",
+            Street = "null",
+            HouseNumber = "null",
+            PostalCode = "null",
+            City = "null",
+            Country = "null",
+            BirthDate = new DateOnly(2002, 12, 12),
+            BirthPlace = "null",
+            PhoneNumber = "null",
+            PlanId = PlanId
+        };
+    }
+}
+
+
+public class Tests : TestBase<App, State>
+{
+    private readonly App _app;
+    private readonly State _state;
+    private readonly VoycarDbContext Context;
+
+
+    // Setup request client
+    public Tests(App app, State state)
+    {
+        this._app = app;
+        this.Context = this._app.Context;
+        this._state = state;
+        State.RoleId = this.Context.Roles.First(r => r.Name == State.RoleName).Id;
+        State.PlanId = this.Context.Plans.First(p => p.Name == State.PlanName).Id;
+    }
+
+
+    [Fact]
+    public async Task Post_Request_ReturnsOk_And_SavesUserInDb()
+    {
+        // Arrange
+        var request = State.CreateValidRequest();
+
+        // Act
+        var httpResponse = await this._app.Client.POSTAsync<R.Endpoint, R.Request>(request);
+
+        // Arrange assertion
+        var userInDb = await this.Context.Users.FirstOrDefaultAsync(user => user.Email == request.Email);
+        var memberInDb = await this.Context.Members.FirstOrDefaultAsync(member => member.Id == userInDb!.MemberId);
+
+        // Assert
+        httpResponse.StatusCode.Should().Be(HttpStatusCode.OK);
+        userInDb.Should().NotBeNull();
+        memberInDb.Should().NotBeNull();
+
+        userInDb.Email.Should().Be(request.Email);
+
+        memberInDb.FirstName.Should().Be(request.FirstName);
+        memberInDb.LastName.Should().Be(request.LastName);
+        memberInDb.Street.Should().Be(request.Street);
+        memberInDb.HouseNumber.Should().Be(request.HouseNumber);
+        memberInDb.PostalCode.Should().Be(request.PostalCode);
+        memberInDb.City.Should().Be(request.City);
+        memberInDb.Country.Should().Be(request.Country);
+        memberInDb.BirthDate.Should().Be(request.BirthDate);
+        memberInDb.BirthPlace.Should().Be(request.BirthPlace);
+        memberInDb.PhoneNumber.Should().Be(request.PhoneNumber);
+        memberInDb.PlanId.Should().Be(request.PlanId);
+        this._state.Id = userInDb.Id; // Save ID for later tests
+    }
+
+
+    [Fact]
+    public async Task Post_Request_ReturnsOk_And_SavesUserInDb_WithLowerInvariantEmail()
+    {
+        // Arrange
+        var request = State.CreateValidRequest();
+        request.Email = "TEST@CAPSLOCK.de";
+
+        // Act
+        var httpResponse = await this._app.Client.POSTAsync<R.Endpoint, R.Request>(request);
+
+        // Arrange assertion
+        var userInDb =
+            await this.Context.Users.FirstOrDefaultAsync(user => user.Email == request.Email.ToLowerInvariant());
+
+        // Assert
+        userInDb!.Email.Should().Be("test@capslock.de");
```

Comment:
Hier dann auch vorher prüfen, dass der user nicht null ist
```suggestion
        userInDb.Should().NotBeNull();
        userInDb!.Email.Should().Be("test@capslock.de");
```